### PR TITLE
=pro #17841 improve PR validator to diff the merge commit for change …

### LINKED ISSUE #17841: pr validation build dependencies doesn't work?
For example, akka-distributed-data was included in this pr validation

https://jenkins.akka.io:8498/job/pr-validator-per-commit-jenkins/3041/consoleFull

which was triggered by pr https://github.com/akka/akka/pull/17839
changes in akka-docs, akka-persistence and akka-sample-persistence-java-lambda

### PR DESCRIPTION
**Pull request by ktoso**
…detection

Resolves #17841
// this will validate _everything_ since it touches project/ that's the diff I want to  see there

### CODE REVIEW

File: project/ValidatePullRequest.scala [hunk 1]
```diff
@@ -143,7 +143,7 @@ object ValidatePullRequest extends AutoPlugin {
 
       // TODO could use jgit
       log.info(s"Comparing [$targetId] with [$prId] to determine changed modules in PR...")
-      val gitOutput = "git diff %s..%s --name-only".format(targetId, prId).!!.split("\n")
+      val gitOutput = "git diff HEAD^ --name-only".!!.split("\n")
```
Comment (2m) [inline project/ValidatePullRequest.scala]: This is great! `targetId` is not used anymore and can be removed with corresponding settings.

**Comment by patriknw** [comment]
LGTM

File: project/ValidatePullRequest.scala (hunk 1 above)
Comment (ktoso) [inline project/ValidatePullRequest.scala]: thanks

**Comment by ktoso** [comment]
Please keep an eye out on PR validation, if works as expected on `master`, it should but better safe than sorry @drewhk @2m

**Pull request merged**